<a href="https://colab.research.google.com/github/nihdin/AppliedEconometrics/blob/master/AE02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
install.packages("wooldridge")
library(wooldridge)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [0]:
data('htv')
head(htv)
colnames(htv)
attach(htv)

wage      abil     educ ne nc west south exper motheduc fatheduc ⋯ ne18 nc18
1 12.019231 5.027738 15   0  0  1    0      9    12       12       ⋯ 1    0   
2  8.912656 2.037170 13   1  0  0    0      8    12       10       ⋯ 1    0   
3 15.514334 2.475895 15   1  0  0    0     11    12       16       ⋯ 1    0   
4 13.333333 3.609240 15   1  0  0    0      6    12       12       ⋯ 1    0   
5 11.070110 2.636546 13   1  0  0    0     15    12       15       ⋯ 1    0   
6 17.482517 3.474334 18   1  0  0    0      8    12       12       ⋯ 1    0   
  south18 west18 urban18 tuit17   tuit18    lwage    expersq ctuit     
1 0       0      1       7.582914  7.260242 2.486508  81     -0.3226714
2 0       0      1       8.595144  9.499537 2.187472  64      0.9043922
3 0       0      1       7.311346  7.311346 2.741764 121      0.0000000
4 0       0      1       9.499537 10.162070 2.590267  36      0.6625338
5 0       0      1       7.311346  7.311346 2.404249 225      0.0000000
6 0       0      1       7.311346  7.311346 2.861201  64      0.0000000

[1] "wage"     "abil"     "educ"     "ne"       "nc"       "west"    
 [7] "south"    "exper"    "motheduc" "fatheduc" "brkhme14" "sibs"    
[13] "urban"    "ne18"     "nc18"     "south18"  "west18"   "urban18" 
[19] "tuit17"   "tuit18"   "lwage"    "expersq"  "ctuit"

In [0]:
r<-range(educ)
print(r[2]-r[1])
summary(educ)

[1] 14


   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   6.00   12.00   12.00   13.04   15.00   20.00 

##### The range of educ variable in the sample is (6,20), i.e. minimum value is 6, maximum value is 20. Therefore the range is 14.

In [0]:
mc2<-nrow(htv[educ == 12,])
mc2
n<-nrow(htv[educ])
p<-(mc2/n)*100
#print(p)
cat("The percentage of men who completed twelfth grade = ", p)

[1] 512

The percentage of men who completed twelfth grade =  41.62602

[1] 13.0374

In [0]:
mean(educ)
mean(motheduc)
mean(fatheduc)
(mean(motheduc)+mean(fatheduc))/2

[1] 13.0374

[1] 12.17805

[1] 12.44715

[1] 12.3126

#####The average education of men in the sample is 13.0374 years, compared to the 12.17 years for mothers and 12.44 for fathers respectively. Even if we took the mean of fathers and mothers together and compared it with that of the men, it is still higher for the men.

In [24]:
m1<-lm(educ~motheduc+fatheduc)
m1
summary(m1)
n


Call:
lm(formula = educ ~ motheduc + fatheduc)

Coefficients:
(Intercept)     motheduc     fatheduc  
     6.9644       0.3042       0.1903  



Call:
lm(formula = educ ~ motheduc + fatheduc)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.2898 -0.9400 -0.4037  1.1239  8.1672 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  6.96435    0.31982  21.776   <2e-16 ***
motheduc     0.30420    0.03193   9.528   <2e-16 ***
fatheduc     0.19029    0.02228   8.539   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.042 on 1227 degrees of freedom
Multiple R-squared:  0.2493,	Adjusted R-squared:  0.248 
F-statistic: 203.7 on 2 and 1227 DF,  p-value: < 2.2e-16


[1] 1230

#The Linear regression model

##### **_$educ$_**= 6.9644 + 0.3042 **_$motheduc$_** + 0.1903 **_$fatheduc$_**

#####$n$ =1230
#####$R^{2}$ = 0.2493

The sample variation that is explained by the mother's and father's education is got by the $R^{2}$ of the fitted model and it comes to around 24.93%.The coefficient on mothers educaation denotes that for every additional grade per education, the concerned man's (adult male) education will be 0.3042 grades higher 

In [25]:
m2<-lm(educ~motheduc+fatheduc+abil)
m2
summary(m2)
n


Call:
lm(formula = educ ~ motheduc + fatheduc + abil)

Coefficients:
(Intercept)     motheduc     fatheduc         abil  
     8.4487       0.1891       0.1111       0.5025  



Call:
lm(formula = educ ~ motheduc + fatheduc + abil)

Residuals:
   Min     1Q Median     3Q    Max 
-5.407 -1.195 -0.199  1.076  7.012 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  8.44869    0.28954  29.180  < 2e-16 ***
motheduc     0.18913    0.02851   6.635 4.87e-11 ***
fatheduc     0.11109    0.01988   5.586 2.85e-08 ***
abil         0.50248    0.02572  19.538  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.784 on 1226 degrees of freedom
Multiple R-squared:  0.4275,	Adjusted R-squared:  0.4261 
F-statistic: 305.2 on 3 and 1226 DF,  p-value: < 2.2e-16


[1] 1230

#The Linear regression model

##### **_$educ$_**= 8.4487 + 0.1891 **_$motheduc$_** + 0.1111 **_$fatheduc$_** + 0.5025 **_$abil$_**

#####$n$ =1230
#####$R^{2}$ = 0.4275

Ability helps explain the variations in education indicating that it is helpful to determine the variation in the dependent variable. The explanatory power of the model increaed to 42.75% after the addition of abil variable in the regression equation.

In [34]:
abil2<-abil*abil
m3<-lm(educ~motheduc+fatheduc+abil+(abil2))
m3
summary(m3)
n
m3df<-as.data.frame(m3[1])
m3df
abil1<-m3df[4,]
abil1sq<-m3df[5,]
min.abil<--abil1/(abil1sq*2)
min.abil


Call:
lm(formula = educ ~ motheduc + fatheduc + abil + (abil2))

Coefficients:
(Intercept)     motheduc     fatheduc         abil        abil2  
     8.2402       0.1901       0.1089       0.4015       0.0506  



Call:
lm(formula = educ ~ motheduc + fatheduc + abil + (abil2))

Residuals:
    Min      1Q  Median      3Q     Max 
-5.2506 -1.1274 -0.1355  1.0223  7.0482 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 8.240226   0.287410  28.671  < 2e-16 ***
motheduc    0.190126   0.028096   6.767 2.03e-11 ***
fatheduc    0.108939   0.019601   5.558 3.35e-08 ***
abil        0.401462   0.030288  13.255  < 2e-16 ***
abil2       0.050599   0.008304   6.093 1.48e-09 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.758 on 1225 degrees of freedom
Multiple R-squared:  0.4444,	Adjusted R-squared:  0.4425 
F-statistic: 244.9 on 4 and 1225 DF,  p-value: < 2.2e-16


[1] 1230

coefficients
(Intercept) 8.24022643  
motheduc    0.19012613  
fatheduc    0.10893866  
abil        0.40146240  
abil2       0.05059901

[1] -3.967098


##### **_$educ$_**= 8.2402 + 0.1901 **_$motheduc$_** + 0.1089 **_$fatheduc$_** + 0.4015 **_$abil$_** + 0.0506 **_$abil^{2}$_**

#####$n$ =1230
#####$R^{2}$ = 0.4444

Differentiating Educ with respect to abil, we get the following
d(Educ)/d(abil) = 0.4015 + 0.1012 **_$abil$_**

$abil^{*}$ = -0.4015/0.1012 = -3.97
Since the value is greater than zero, we have a minimum

In [35]:
nrow(htv[abil <= min.abil,])

[1] 15

#####This is important since there are only 15 individuals who will have a higher expected education level than individuals to slightly right of -3.97. In other words, there are only 15 values of the equation that are counter-intuitive.